packages/medusa/src/services/region.ts

```diff
@@ -225,19 +264,19 @@ class RegionService extends BaseService {
       this.validateTaxRate_(region.tax_rate)
     }
 
-    if (region.countries) {
-      region.countries = await Promise.all(
-        region.countries.map((countryCode) =>
-          this.validateCountry_(countryCode, id)
+    if ((region as T & { countries: Country[] }).countries) {
```

Comment:
suggestion: Same as on of the previous, we should create a new object on which we assign the values and then return it. It will also fix the fact that the type defer since it is not the same object

Reply:
comment: It also does not return T but a new type of object apparently